dev/integration_tests/display_cutout_rotation/integration_test/display_cutout_test.dart

```diff
@@ -0,0 +1,150 @@
+            'cutout should start at the left, does the test device have a '
+            'camera cutout or window inset?',
+      );
+    });
+
+    testWidgets('cutout handles rotation', (tester) async {
+      await moreReliableSetOrentations(tester, DeviceOrientation.portraitUp);
+      const widgetUnderTest = MyApp();
+      // Load app widget.
+      await tester.pumpWidget(widgetUnderTest);
+      BuildContext context = tester.element(find.byType(Text));
+      List<DisplayFeature> displayFeatures = getCutouts(tester, context);
+      // Test is expecting one cutout setup in the test harness.
+      expect(
+        displayFeatures.length,
+        1,
+        reason: 'Single cutout display feature expected',
+      );
+      // Verify that app code thinks there is a top cutout.
+      expect(
+        displayFeatures[0].bounds.top,
+        0,
+        reason:
+            'cutout should start at the top, does the test device have a '
+            'camera cutout or window inset?',
+      );
+      await moreReliableSetOrentations(tester, DeviceOrientation.landscapeLeft);
+      await tester.pumpWidget(widgetUnderTest);
+
+      // Requery for display features after rotation.
+      context = tester.element(find.byType(Text));
+      displayFeatures = getCutouts(tester, context);
+      // Test is expecting one cutout setup in the test harness.
+      expect(
+        displayFeatures.length,
+        1,
+        reason: 'Single cutout display feature expected',
+      );
+      expect(
+        displayFeatures[0].bounds.left,
+        0,
+        reason: 'cutout should start at the left or handle camera',
+      );
+    });
+
+    tearDown(() {
+      // After each test reset to device perfered orientations to avoid
+      // test pollution.
+      SystemChrome.setPreferredOrientations([]);
+    });
+  });
+}
+
+/*
+ * Force rotation then poll to ensure rotation has happened.
+ *
+ * Rotations have an async communication to engine which then has an async
+ * communication to the android operating system.
+ */
+Future<void> moreReliableSetOrentations(
+  WidgetTester tester,
+  DeviceOrientation orientation,
+) async {
+  SystemChrome.setPreferredOrientations([orientation]);
+  Orientation expectedOrientation;
+  switch (orientation) {
+    case DeviceOrientation.portraitUp:
+    case DeviceOrientation.portraitDown:
+      expectedOrientation = Orientation.portrait;
+      break;
+    case DeviceOrientation.landscapeRight:
+    case DeviceOrientation.landscapeLeft:
+      expectedOrientation = Orientation.landscape;
+      break;
+  }
+  do {
+    BuildContext context = tester.element(find.byType(Text));
+    if (expectedOrientation == MediaQuery.of(context).orientation) {
+      break;
+    }
+    await tester.pumpAndSettle();
+  } while (true);
+}
+
+List<DisplayFeature> getCutouts(WidgetTester tester, BuildContext context) {
+  List<DisplayFeature> displayFeatures = MediaQuery.of(context).displayFeatures;
+  displayFeatures.retainWhere(
```

Comment:
I am not certain/would not want to be certain that `displayFeatures` is always a copy, and `retainWhere` (confusingly), is a mutation. To not think about this, let's return an `Iterable<DisplayFeature>` and just use `.where`.